homeassistant/components/lastfm/sensor.py

```diff
@@ -35,95 +30,49 @@
+    async_add_entities(
+        (LastFmSensor(coordinator, user) for user in config[CONF_USERS]), True
+    )
 
-    entities = []
-    for username in users:
-        try:
-            lastfm_api.get_user(username).get_image()
-            entities.append(LastfmSensor(username, lastfm_api))
-        except WSError as error:
-            _LOGGER.error(error)
-            return
 
-    add_entities(entities, True)
-
-
-class LastfmSensor(SensorEntity):
+class LastFmSensor(CoordinatorEntity[LastFmUpdateCoordinator], SensorEntity):
     """A class for the Last.fm account."""
 
     _attr_attribution = "Data provided by Last.fm"
     _attr_icon = "mdi:radio-fm"
 
-    def __init__(self, user, lastfm_api):
+    def __init__(self, coordinator: LastFmUpdateCoordinator, user: str) -> None:
         """Initialize the sensor."""
-        self._unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
-        self._user = lastfm_api.get_user(user)
-        self._name = user
-        self._lastfm = lastfm_api
-        self._state = "Not Scrobbling"
-        self._playcount = None
-        self._lastplayed = None
-        self._topplayed = None
-        self._cover = None
-
-    @property
-    def unique_id(self):
-        """Return the unique ID of the sensor."""
-        return self._unique_id
-
-    @property
-    def name(self):
-        """Return the name of the sensor."""
-        return self._name
-
-    @property
-    def native_value(self):
-        """Return the state of the sensor."""
-        return self._state
-
-    def update(self) -> None:
-        """Update device state."""
-        self._cover = self._user.get_image()
-        self._playcount = self._user.get_playcount()
-
-        if recent_tracks := self._user.get_recent_tracks(limit=2):
-            last = recent_tracks[0]
-            self._lastplayed = f"{last.track.artist} - {last.track.title}"
-
-        if top_tracks := self._user.get_top_tracks(limit=1):
-            top = str(top_tracks[0])
-            if (toptitle := re.search("', '(.+?)',", top)) and (
-                topartist := re.search("'(.+?)',", top)
-            ):
-                self._topplayed = f"{topartist.group(1)} - {toptitle.group(1)}"
-
-        if (now_playing := self._user.get_now_playing()) is None:
-            self._state = STATE_NOT_SCROBBLING
-            return
-
-        self._state = f"{now_playing.artist} - {now_playing.title}"
-
-    @property
-    def extra_state_attributes(self):
-        """Return the state attributes."""
-        return {
-            ATTR_LAST_PLAYED: self._lastplayed,
-            ATTR_PLAY_COUNT: self._playcount,
-            ATTR_TOP_PLAYED: self._topplayed,
-        }
-
-    @property
-    def entity_picture(self):
-        """Avatar of the user."""
-        return self._cover
+        super().__init__(coordinator)
+        self._attr_unique_id = hashlib.sha256(user.encode("utf-8")).hexdigest()
+        self._attr_name = f"lastfm_{user}"
```

Comment:
```suggestion
        self._attr_name = user
```

Reply:
I am still wondering why user would be better than lastfm_user. because currently my entity is sensor.joostlek. But for Twitch the sensor name is also the username, which would be sensor.joostlek and that'd be a collision. How can we safely migrate this?

Reply:
You should worry about that as a standalone PR :)
For now in refactoring you try to keep identical behavior.
Maybe "LastFM user" would be better... but for a specific PR